src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -75,13 +76,11 @@ def oauth2_client_credentials_tokens(
         >>> transfer_token = transfer_token_info["access_token"]
         """
         log.info("Fetching token(s) using client credentials")
-        requested_scopes = requested_scopes or DEFAULT_REQUESTED_SCOPES
-        # convert scopes iterable to string immediately on load
-        if not isinstance(requested_scopes, str):
-            requested_scopes = " ".join(requested_scopes)
-
+        requested_scopes_string: str = scopes.MutableScope.scopes2str(
+            requested_scopes or DEFAULT_REQUESTED_SCOPES
+        )
```

Comment:
If I'm reading correctly, since the comparison is falsey rather than with `None` then a user requesting an empty set of scopes would get the default set of scopes. I highly doubt there's any reasonable use case for a user to want to do that, but I would find it a little surprising that it requests four scopes when I explicitly requested no scopes—rather than, say, throwing an error.

This is not a change from the existing behavior and in any case I don't think it warrants a change, but wanted to mention in case there was some nuance that it might be helpful for me to know!

Reply:
Great catch!

Reply:
I think there are a couple of locations which have this behavior on scopes.
Let me merge this PR to get the main improvement in and take a look at putting this behavior (`scopes2str` + error handling for `""` and maybe other bad values) into a helper somewhere.